packages/protocol/contracts/stability/PairManager.sol

```diff
@@ -0,0 +1,290 @@
+    broker = _broker;
+    emit BrokerUpdated(_broker);
+  }
+
+  /**
+   * @notice Creates a new pair using the given parameters.
+   * @param pair The information required to create the pair.
+   * @return pairId The id of the newly created pair.
+   */
+  function createPair(Pair calldata pair) external onlyOwner returns (bytes32 pairId) {
+    require(address(pair.mentoExchange) != address(0), "Mento exchange must be set");
+    require(address(pair.stableAsset) != address(0), "Stable asset must be set");
+    require(address(pair.collateralAsset) != address(0), "Collateral asset must be set");
+
+    Pair memory pairInfo = pair;
+
+    pairId = keccak256(
+      abi.encodePacked(
+        IERC20Metadata(pairInfo.stableAsset).symbol(),
+        IERC20Metadata(pairInfo.collateralAsset).symbol(),
+        pairInfo.mentoExchange.name()
+      )
+    );
+    require(!isPair[pairId], "A pair with the specified assets and exchange exists");
+
+    validatePairInfo(pairInfo);
+
+    // TODO: getUpdatedBuckets Function sig should change. May not need amounts
+    // but instead pass minSupplyForStableBucketCap & stableBucketMaxFraction
+    (uint256 tokenInBucket, uint256 tokenOutBucket) = pairInfo.mentoExchange.getUpdatedBuckets(
+      pairInfo.stableAsset,
+      pairInfo.collateralAsset,
+      0,
+      0,
+      pairId
+    );
+
+    pairInfo.stableBucket = tokenInBucket;
+    pairInfo.collateralBucket = tokenOutBucket;
+
+    // TODO: https://github.com/mento-protocol/mento-general/issues/60
+    // Update allocated collateral fraction
+    // allocatedCollateralFractions[pairInfo.collateralAsset] = allocatedCollateralFractions[
+    //   pairInfo.collateralAsset
+    // ].add(pairInfo.collateralBucketFraction);
+
+    // // Update allocated stable bucket fraction
+    // allocatedStableFractions[pairInfo.stableAsset] = allocatedStableFractions[pairInfo.stableAsset]
+    //   .add(pairInfo.stableBucketMaxFraction);
+
+    pairs[pairId] = pairInfo;
+    isPair[pairId] = true;
+
+    emit PairCreated(
+      pairInfo.stableAsset,
+      pairInfo.collateralAsset,
+      address(pairInfo.mentoExchange)
+    );
+  }
+
+  /**
+   * @notice Destroys a pair with the given parameters if it exists and frees up
+   *         the collateral and stable allocation it was using.
+   * @param stableAsset The stable asset of the pair.
+   * @param collateralAsset The collateral asset of the pair.
+   * @return destroyed A boolean indicating whether or not the pair was successfully destroyed.
+   */
+  function destroyPair(address stableAsset, address collateralAsset, IMentoExchange mentoExchange)
+    external
+    onlyOwner
+    returns (bool destroyed)
+  {
+    require(stableAsset != address(0), "Stable asset address must be specified");
+    require(collateralAsset != address(0), "Collateral asset address must be specified");
+    require(address(mentoExchange) != address(0), "Mento exchange must be set");
```

Comment:
Wonder if this is necessary.

Reply:
Probably not. Was thinking of how to guarantee MentoExchange.name() would always return something. But we can remove this, worst case the pair isn't found then delete fails.